src/Symfony/Bridge/Doctrine/Middleware/ConnectionKeepAliveDriver.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Middleware;
+
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Driver\Connection as DriverConnection;
+use Doctrine\DBAL\Driver\Middleware\AbstractDriverMiddleware;
+
+final class ConnectionKeepAliveDriver extends AbstractDriverMiddleware
+{
+    public function __construct(
+        Driver $driver,
+        private readonly \ArrayObject $connectionTimings,
+        private readonly array $timeToLiveByConnection,
+        private readonly string $connectionName
+    ) {
+        parent::__construct($driver);
+    }
+
+    public function connect(array $params): DriverConnection
+    {
+        $connection = parent::connect($params);
+        $this->connectionTimings->append(['expiry' => time() + $this->timeToLiveByConnection[$this->connectionName], 'name' => $this->connectionName]);
```

Comment:
```suggestion
        $this->connectionEols[$this->connectionName] = time() + $this->timeToLiveByConnection[$this->connectionName]);
```

I think this is the wrong data structure: you will have as many lines in there as SQL queries. I think you should use the connection name as a key, and overwrite the limit every time a query happens. Let me know what I'm missing